### ISSUE TITLE
Incremental Prompting: Episodic Memory Prompt for Lifelong Event Detection

### ISSUE THREAD

**Issue by HYUNMIN-HWANG** [One-Day-One-Paper/One-Day-One-Paper#274]
# Incremental Prompting: Episodic Memory Prompt for Lifelong Event Detection

기존 문제 : class-incremental learning을 할 때 catastrophic forgetting 문제가 발생함, 이를 해결하기 위해 fine-tuning 하다보면 overfitting 문제가 발생했다.

# Episodic Memory Prompts (EMP)

- incremental prompting framework
- preserve the learned task-specific knowledge

## Span-based Event Detection

- Encode BERT
- $x^t_{1:L} = BERT(x^t_{1:L})$
- each span

$$
\bar{x}^t→ concat(\bar{x}starting, \bar{x} ending  token)→MLP→h^t_{span}→linear→p^t
$$

- cross entropy

![Untitled](Incremental%20Prompting%20Episodic%20Memory%20Prompt%20for%20L%20539bc1933b024353bc0684608ed39ae4/Untitled.png)

## Episodic Memory Prompting

- task $D_t$
- event set $C_t$
- accumulate prompts : $I^t=[C^1,...,C^t]$
- original logits : concatenate $x^t_{1:L}$ 와 $I^t$ → encode BERT → contextual representations $[\tilde{x}^t_{1:L};\tilde{I}^t]$ → MLP → $\tilde{h}^t_{span}$→ linear → $\tilde{p}^t$
- probability distribution over all prompts : $\tilde{p}^t_c = MLP(\tilde{I}^t)\cdot\tilde{h}^t_{span}$
- predict the event type label for each span

![Untitled](Incremental%20Prompting%20Episodic%20Memory%20Prompt%20for%20L%20539bc1933b024353bc0684608ed39ae4/Untitled%201.png)

- initialize a new prompt $C^{l+1}$
- Next task incrementally $I^{t+1}=[I^t;C^{t+1}]$

## Lifelong Learning

(1) Experience Replay (ER) : training sample 중 20개 훈련데이터 셋에 관련된 loss

![Untitled](Incremental%20Prompting%20Episodic%20Memory%20Prompt%20for%20L%20539bc1933b024353bc0684608ed39ae4/Untitled%202.png)

(2) Knowledge Distillation (KD) : prediction-level 과 feature-level distillation 사용

![Untitled](Incremental%20Prompting%20Episodic%20Memory%20Prompt%20for%20L%20539bc1933b024353bc0684608ed39ae4/Untitled%203.png)

### Optimization

![Untitled](Incremental%20Prompting%20Episodic%20Memory%20Prompt%20for%20L%20539bc1933b024353bc0684608ed39ae4/Untitled%204.png)

---

- DATASET : ACE05-EN & MAVEN
- 5 sequence of tasks

## Results

![Untitled](Incremental%20Prompting%20Episodic%20Memory%20Prompt%20for%20L%20539bc1933b024353bc0684608ed39ae4/Untitled%205.png)

- Task1은 catastrophic 문제가 없는 상태 : 이때도 EMP 성능 좋다
- KCN은 EMP가 없는 버전 : KCN보다 성능이 좋다
- BERT-ED : KCN에서 experience replay and knowledge distillation이 없는 버전, KCN 보다 성능이 안 좋음
- 여전히 task1과 task5 사이의 간극은 있다.
